server/pulp/server/async/tasks.py

```diff
@@ -507,3 +514,5 @@ def cleanup_old_worker(*args, **kwargs):
     """
     name = kwargs['sender'].hostname
     _delete_worker(name, normal_shutdown=True)
+    # Recreate a new working directory for worker that is starting now
+    common_utils.create_worker_working_directory(name)
```

Comment:
@beav cleanup_old_worker has a decorator that connects it to a worker_init signal.  This method is run when a worker starts.